### PR TITLE
#S9EuF0ch Logout an Auth User

### PR DESCRIPTION
**Pull request by karigirwa**
### **What about this PR?**
This PR is for logout an Authenticated User

### **Description of Task to be completed**
   - Authenticated users are able to logout.
   - Appropriate error messages are displayed on unsuccessful attempts
   - Skeleton UIs/screen are/is shown when the page loads/is loading.
   - The app layouts are displayed
   - The page is responsive. 
   
  **How should this be manually tested?**

- clone this repo and do git checkout `ch-logout-auth-S9EuF0ch`

- First navigate to http://localhost:5001/login and Login 

- To the dashboard page, there is a Logout button

- Click Logout button to Logout from account

### **Relevant pivots tracker**
[#9EuF0ch](https://trello.com/c/S9EuF0ch/30-logout-from-account)

**Screenshot**

logout 1

### CODE REVIEW

**Review by ceelogre — CHANGES_REQUESTED** [review]
Your description is not complete, how should it be tested?

**Comment by ceelogre** [comment]
Since signup is not complete, which credentials should be used to signin?

**Comment by karigirwa** [comment]
as signup is not completed, the user can sign in using email:" admin@phantom.com" and Password: "admin"

**Comment by ceelogre** [comment]
Attempting to signout throws an error in the console.


Capture d’écran 2020-12-16 à 10 24 51

**Comment by karigirwa** [comment]
Test it again

File: src/redux/actions/logoutAction.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { LOGOUT_REQUEST, LOGOUT_SUCCESS, LOGOUT_ERROR } from '../types/logoutTypes'
+import axios from 'axios'
+
+export const logout = (history) => async(dispatch) => {
+    try {
+        dispatch(logoutRequest());
+        // console.log('Request')
+        const res = await axios.get('https://phantom-caba-ft-logout--az7r73.herokuapp.com/api/v1/auth/logout')
+        localStorage.removeItem('token')
+        localStorage.clear()
```
Comment (ceelogre) [inline src/redux/actions/logoutAction.js]: Why do you have to clear the localstorage? you might be accidentally removing some other pieces of information stored that are important.

File: src/redux/reducers/logoutReducer.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    data: {},
+    error:''
+};
+
+const logoutReducer = (state = initialState, action) => {
+  switch (action) {
+    case actionTypes.LOGOUT_REQUEST:
+      return { 
+          ...state, 
+          loading: 'block'
```
Comment (ceelogre) [inline src/redux/reducers/logoutReducer.js:14]: What's the use of this?

File: src/__tests__/logout.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+const AppComponent = () => {
+    return render(
+        <Provider store={store}>
+            <Router>
+                <Logout />
+            </Router>
+        </Provider>
+    )
+}
```
Comment (ceelogre) [inline src/__tests__/logout.test.js]: This component can be imported not necessarily re-created.

**Review by ceelogre — CHANGES_REQUESTED** [review]

File: src/redux/reducers/logoutReducer.js (hunk 2 above)
Comment (karigirwa) [reply src/redux/reducers/logoutReducer.js:14]: this is used because most of actions in redux need Request action

Comment (ceelogre) [reply src/redux/reducers/logoutReducer.js:14]: The question is: how does your app react when you change `block` to `none` ?

**Review by ceelogre — APPROVED** [review]

**Pull request merged**